packages/open-next/src/adapters/server-adapter.ts

```diff
@@ -131,30 +136,47 @@ function initializeNextjsRequireHooks(config: any) {
   // WORKAROUND: Set `__NEXT_PRIVATE_PREBUNDLED_REACT` to use prebundled React — https://github.com/serverless-stack/open-next#workaround-set-__next_private_prebundled_react-to-use-prebundled-react
   if (!isNextjsVersionAtLeast("13.1.3")) return;
   overrideDefault();
-  overrideReact(config);
+  overrideReact(config, node_react);
+}
+
+function getRoutePattern(url: string): string | null {
+  let routePattern: string | null = null;
+  routesManifest.staticRoutes.forEach((route) => {
+    if (new RegExp(route.regex).test(url)) {
+      // We need to add `/page` to the end of the route pattern because that's how it is in the appPathsManifest
+      routePattern = `${route.page}/page`;
+    }
+  });
+  if (routePattern) return routePattern;
+  routesManifest.dynamicRoutes.forEach((route) => {
+    if (new RegExp(route.regex).test(url)) {
+      routePattern = `${route.page}/page`;
+    }
+  });
```

Comment:
Do we need to also handle `routesManiftest.dataRoutes` here?

Reply:
Not useful here, we just check if it's an app and app doesn't have separate data route